core/src/main/java/com/google/cloud/sql/core/DefaultAccessTokenSupplier.java

```diff
@@ -0,0 +1,199 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.sql.core;
+
+import com.google.api.client.auth.oauth2.Credential;
+import com.google.api.client.http.HttpRequestInitializer;
+import com.google.auth.Credentials;
+import com.google.auth.http.HttpCredentialsAdapter;
+import com.google.auth.oauth2.AccessToken;
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.common.annotations.VisibleForTesting;
+import java.io.IOException;
+import java.time.Duration;
+import java.util.Date;
+import java.util.Optional;
+
+/**
+ * DefaultAccessTokenSupplier produces access tokens using credentials produced by this connector's
+ * configured HttpRequestInitializer.
+ */
+class DefaultAccessTokenSupplier implements AccessTokenSupplier {
```

Comment:
Is there a test class missing from this PR?

Reply:
No: core/src/test/java/com/google/cloud/sql/core/DefaultAccessTokenSupplierTest.java